datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireViewModelTest.kt

```
@@ -3434,20 +3476,22 @@ class QuestionnaireViewModelTest {
           )
         }
       val viewModel = createQuestionnaireViewModel(questionnaire, enableReviewPage = false)
-      viewModel.runViewModelBlocking {
-        viewModel.goToNextPage()
 
-        assertThat(
-            (viewModel.questionnaireStateFlow.value as DisplayMode.EditMode)
-              .pagination.showReviewButton
-          )
-          .isFalse()
-      }
+      viewModel.goToNextPage()
+      val collectJob =
+        launch(mainDispatcherRule.testDispatcher) { viewModel.questionnaireStateFlow.collect() }
+
+      assertThat(
+          (viewModel.questionnaireStateFlow.value.displayMode as DisplayMode.EditMode)
+            .pagination.showReviewButton
+        )
+        .isFalse()
+      collectJob.cancel()
```

you could have a private helper function:

```
private fun something(block: {}) {
  val collectJob = ...
  block.run()
  collectJob.canel()
}
```

---

Implemented this! Great suggestion as this cleaned up the code. I also added a test case where we do an incorrect cast just to check that an exception is raised and is not swallowed like before